Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard_test.py

```diff
@@ -0,0 +1,647 @@
+        - Case B: Score received for 2021-07-01 to 2021-07-31 on daily basis
+        - Case C: Score received for 2021-07-01 to 2021-07-31 on weekly basis
+    """
+
+    mocker.patch.object(client, "get_company_historical_factor_scores", return_value=historical_factor_score_mock)
+
+    cmd_res: CommandResults = company_history_factor_score_get_command(
+        client=client,
+        domain=domain,
+        _from=_from,
+        to=to,
+        timing=timing
+    )
+
+    factor_scores = cmd_res.outputs
+    assert factor_scores == historical_factor_score_mock.get("entries")
+
+
+@pytest.mark.parametrize("username, change_direction, score_types, target, portfolios", grade_alert_test_input)
+def test_create_grade_change_alert(mocker, username, change_direction, score_types, target, portfolios):
+    """
+    Given:
+        - A username
+        - Direction change
+        - Score type(s)
+        - Target or Portfolio(s)
+    When:
+        - Case A: Username is valid, rising grade, overall score type, to portfolio
+        - Case B: Username is valid, dropping grade, overall and cubit score score types, to portfolio
+        - Case C: Username is valid, rising grade, application security score type to my scorecard
+        - Case D: Both portfolio and target are specified
+        - Case E: Neither portfolio and target are specified
+    Then:
+        - Case A: Alert created
+        - Case B: Alert created
+        - Case C: Alert created
+        - Case D: DemistoException thrown
+        - Case E: DemistoException thrown
+    """
+
+    mocker.patch.object(client, "create_grade_change_alert", return_value=create_grade_alert_mock)
+
+    if target and portfolios:
+        with pytest.raises(DemistoException) as exc:
+            alert_grade_change_create_command(
+                client=client,
+                username=username,
+                change_direction=change_direction,
+                score_types=score_types,
+                target=target,
+                portfolios=portfolios
+            )
+
+        assert "Both 'portfolio' and 'target' argument have been set" in str(exc.value)
+    elif not target and not portfolios:
+        with pytest.raises(DemistoException) as exc:
+            alert_grade_change_create_command(
+                client=client,
+                username=username,
+                change_direction=change_direction,
+                score_types=score_types,
+                target=target,
+                portfolios=portfolios
+            )
+
+        assert "Either 'portfolio' or 'target' argument must be given" in str(exc.value)
+    else:
+
+        cmd_res: CommandResults = alert_grade_change_create_command(
+            client=client,
+            username=username,
+            change_direction=change_direction,
+            score_types=score_types,
+            target=target,
+            portfolios=portfolios
+        )
+
+        assert cmd_res.outputs == create_grade_alert_mock.get("id")
```

Comment:
The cases are different because of the arguments used:

```
        - Case A: rising grade, overall score type, to portfolio
        - Case B: dropping grade, overall and cubit score score types, to portfolio
        - Case C: rising grade, application security score type to my scorecard
```

The response from the API only returns the same response (namely the ID of the generated alert). In this case, the mock will return the same ID (`1`) for all good test cases and a `DemistoException` for the bad cases.

Reply:
I agree, so what is the advantage of having 3 different arguments? the outcome will stay the same (intentionally) and it is not improving cover.

Reply:
Got it. Removed test cases when response is identical in both `test_create_score_change_alert` and `test_create_grade_change_alert`.